test/smoke/sampler/src/test/test.js

```diff
@@ -0,0 +1,143 @@
+  'rating_plan': spanConfig.rating_plan,
+  'pricing_plan': spanConfig.pricing_plan
+};
+
+const createEventBuilder = () => {
+  const _getEvent = (time) => ({
+    timestamp: time,
+    organization_id: spanConfig.organization_id,
+    space_id: spanConfig.space_id,
+    consumer_id: spanConfig.consumer_id,
+    resource_id: spanConfig.resource_id,
+    plan_id: spanConfig.plan_id,
+    resource_instance_id: spanConfig.resource_instance_id,
+    measured_usage: []
+  });
+
+  return {
+    createStartEvent: (time) => _getEvent(time),
+    createStopEvent: (time) => omit(_getEvent(time), 'measured_usage')
+  };
+};
+
+
+const getMonthlySummaryValue = (report) => {
+  const resourceIndex = 0;
+  const aggrUsageMetricIndex = 0;
+  const monthWindowIndex = 4;
+  const currentMonthIndex = 0;
+  const parsedReport = JSON.parse(report);
+
+  // when run for the first time, there is no data in report
+  if(parsedReport.resources.length < 1)
+    return 0;
+
+  return parsedReport
+    .resources[resourceIndex]
+    .aggregated_usage[aggrUsageMetricIndex]
+    .windows[monthWindowIndex][currentMonthIndex]
+    .summary;
+};
+
+const log = (msg) => console.log(`${moment.utc().toDate()}: ${msg}`);
+
+const startTokens = async() => {
+  const startSystemToken = util.promisify(systemToken.start);
+  const startSamplerToken = util.promisify(samplerToken.start);
+
+  await startSystemToken();
+  await startSamplerToken();
+};
+
+const getToken = (token) => env.secured ? token() : undefined;
+
+describe('Sampler smoke test', function() {
+  let receiverClient;
+  let reportingClient;
+  let eventBuilder;
+
+  this.timeout(env.totalTimeout);
+  setEventuallyPollingInterval(env.pollInterval);
+  setEventuallyTimeout(env.totalTimeout);
+
+  const createMappingGracefully = async (receiverClient, mapping) => {
+    try {
+      await receiverClient.createMappings(mapping);
+    } catch(e) {
+      expect(e.message).to.equal(new ConflictError().message);
+    }
+  };
+
+  const stopSamplingGracefully = async (receiverClient, event) => {
+    try {
+      await receiverClient.stopSampling(event);
+    } catch(e) {
+      expect(e.message).to.equal(new UnprocessableEntityError().message);
+    }
+  };
+
+  before(async() => {
+    if(env.secured)
+      await startTokens();
+
+    receiverClient = new ReceiverClient(env.receiverUrl, { getHeader: () => getToken(samplerToken) }, env.skipSSL);
+    reportingClient = new ReportingClient(env.reportingUrl, { getHeader: () => getToken(systemToken) }, env.skipSSL);
+    eventBuilder = createEventBuilder();
+  });
+
+  after(async() => {
+    log('Stopping sampling in case of test error ...');
+    await stopSamplingGracefully(receiverClient, eventBuilder.createStopEvent(moment.now()));
+  });
+
+  it('samples usage to Abacus successfully', async () => {
+    log('Creating services mappings ...');
+    await createMappingGracefully(receiverClient, mapping);
+
+    log('Cleaning up in case of previous test error ...');
```

Comment:
Shouldn't this be in the `before` or `beforeEach` section? If we ever do another `it` we won't need to copy-paste it.

Reply:
New context with `beforeEach` was pushed